From fd7f253649acc3cca498951f71533f012cef1b9e Mon Sep 17 00:00:00 2001
From: Hugo Locurcio <hugo.locurcio@hugo.pro>
Date: Mon, 18 Mar 2019 14:37:25 +0100
Subject: [PATCH] Add support for linking using LLD on X11

LLD is often faster than GNU ld and gold, resulting in a better
development experience.

This closes #15364.

----
Modified patch to backport to 2.1.7_rc
Modified by Orson Teodoro <orsonteodoro@hotmail.com>
Date: Jun 1, 2021

diff -urp godot-89e531d223ef189219e266cc61ea79a7dd2d5f54.orig/platform/x11/detect.py godot-89e531d223ef189219e266cc61ea79a7dd2d5f54/platform/x11/detect.py
--- godot-89e531d223ef189219e266cc61ea79a7dd2d5f54.orig/platform/x11/detect.py	2020-10-01 10:50:39.000000000 -0700
+++ godot-89e531d223ef189219e266cc61ea79a7dd2d5f54/platform/x11/detect.py	2021-06-01 10:39:49.589138643 -0700
@@ -52,6 +52,7 @@ def get_opts():
 
     return [
         ('use_llvm', 'Use llvm compiler', 'no'),
+        ('use_lld', 'Use the LLD linker', 'no'),
         ('use_static_cpp', 'Link libgcc and libstdc++ statically for better portability', 'no'),
         ('use_sanitizer', 'Use llvm compiler sanitize address', 'no'),
         ('use_leak_sanitizer', 'Use llvm compiler sanitize memory leaks', 'no'),
@@ -116,6 +117,12 @@ def configure(env):
         env.Append(CCFLAGS=['-flto'])
         env.Append(LINKFLAGS=['-flto'])
 
+    if (env['use_lld'] == "yes"):
+        if (env['use_llvm'] == "yes"):
+            env.Append(LINKFLAGS=['-fuse-ld=lld'])
+        else:
+            print("Using LLD with GCC is not supported yet, try compiling with 'use_llvm=yes'.")
+            sys.exit(255)
 
     env.Append(CCFLAGS=['-pipe'])
     env.Append(LINKFLAGS=['-pipe'])
Only in godot-89e531d223ef189219e266cc61ea79a7dd2d5f54/platform/x11: detect.py.orig
Only in godot-89e531d223ef189219e266cc61ea79a7dd2d5f54/platform/x11: detect.py.rej
